perma_web/perma/tests/test_views_auth.py

```diff
@@ -8,152 +7,145 @@
-        self.assertFalse(AccessAttempt.objects.filter(username=self.email).exists())
-
-        # verify you get the normal form errors, not the lockout page,
-        # if you fail again
-        response = self.attempt_login(self.email, self.wrong_password, expect_success=False)
-        self.assertContains(response, 'class="field-error"', status_code=200)
-
-        # verify you CAN login with the new password
-        self.log_in_user(user=self.email, password=self.new_password)
+
+from conftest import TEST_USER_PASSWORD, randomize_capitalization, submit_form
+
+
+def attempt_login(perma_client, username, password, expect_success=True):
+    assert '_auth_user_id' not in perma_client.session
+
+    response = perma_client.post(reverse('user_management_limited_login'),
+                                 {'username': username,
+                                  'password': password})
+    if expect_success:
+        assert response.status_code == 302
+        assert 'login' not in response['Location']
+        assert '_auth_user_id' in perma_client.session
+    else:
+        assert '_auth_user_id' not in perma_client.session
+    return response
+
+def test_login(perma_client, link_user):
+    """
+    Test the login form
+    We should get redirected to the create page
+    """
+    # Login through our form and make sure we get redirected to our create page,
+    # no matter how the email address is capitalized
+    assert LinkUser.objects.filter(email__iexact=link_user.email).count() == 1
+
+    attempt_login(perma_client, link_user.email, TEST_USER_PASSWORD)
+    perma_client.logout()
+    attempt_login(perma_client, link_user.email.upper(), TEST_USER_PASSWORD)
+    perma_client.logout()
+    attempt_login(perma_client, link_user.email.title(), TEST_USER_PASSWORD)
+    perma_client.logout()
+    attempt_login(perma_client, randomize_capitalization(link_user.email), TEST_USER_PASSWORD)
+
+def test_deactived_user_login(perma_client, deactivated_user):
+    submit_form(perma_client, 'user_management_limited_login',
+                    data = {'username': deactivated_user.email,
+                            'password': TEST_USER_PASSWORD},
+                    success_url=reverse('user_management_account_is_deactivated'))
+    assert '_auth_user_id' not in perma_client.session
+
+def test_unactived_user_login(perma_client, unactivated_user):
+    submit_form(perma_client, 'user_management_limited_login',
+                        data = {'username': unactivated_user.email,
+                                'password': 'pass'},
+                        success_url=reverse('user_management_not_active'))
+    assert '_auth_user_id' not in perma_client.session
+
+def test_logout(perma_client, link_user):
+    """
+    Test our logout link
+    """
+
+    # Login with our client and logout with our view
+    attempt_login(perma_client, link_user.email, TEST_USER_PASSWORD)
+    assert '_auth_user_id' in perma_client.session
```

Comment:
`attempt_login` already checks this assertion, right?

I think you can remove that assertion every time you call `attempt_login`,